### DISCUSSION TITLE
What is the difference between 64Bit & non-64Bit engines?

### DISCUSSION THREAD

**Discussion by BenceSzalai** [The-Wineskin-Project/WineskinServer#429] [Q&A] [answered]
I cannot find a comprehensive answer about what the two are differ in.

My guess would be that non-64Bit engines were supposed to run only 32-bit windows apps, while 64Bit engines were supposed to run both 32- and 64-bit windows apps. This seems to match that - apparently - from CX v22 there is no more non-64Bit engine version.

However when I'm trying to run my 32-bit Windows app which works fine in CX21 under CX64Bit22 it fails, suggesting that 64Bit engines may not be able to run 32-bit windows apps...?

What is the truth? How do you choose the right engine?

**Comment by Gcenx** [comment, marked as answer]
Indeed non 64Bit engines would only generate a 32Bit wineprefix, but CX22 I’ve been unable to have wine32on64 nor wine64 generate valid 32Bit wineprefixes.

I can’t say I’d CX23 will or won’t support generating valid 32Bit wineprefixes but eveb if it’s possible CX23 onward is using upstreams new wow64 so regardless the Engine would still essentially be 64Bit.

Unfortunately regressions happen, it can be that CX22 added a stub that your app is attempting to use or perior wine didn’t provide a builtin dll but now does and that builtin dll might not be good enough to run your app etc.

**Reply by mohd-akram** [reply]
Is there a way to get a 32-bit prefix currently? I'm trying to install ie8 via winetricks and it complains about requiring a 32-bit prefix. I noticed that older engines have been removed from the list as well. Perhaps if it's not possible with current engines, the latest 32-bit supporting engine could also be kept in the list.

**Reply by Gcenx** [reply]
@mohd-akram there's currently no way to create a 32Bit wineprefix with the current supported versions of wine I'm providing.

The prior engines were removed as they're no longer supported.

**Comment by BenceSzalai** [comment]
Thank you! So long story short 64Bit engine supposed to be able to run both 32- and 64-bit apps. If they don't, that is probably due to other issue, not the bitness of the app vs. engine. Right?

**Reply by Gcenx** [reply]
Pretty much, sometimes it’s wine regressions and sometimes the apps simply don’t like wow64 this is true even on windows.
